src/librustc/ty/sty.rs

```diff
@@ -127,9 +127,26 @@ pub enum TyKind<'tcx> {
 
     /// The anonymous type of a function declaration/definition. Each
     /// function has a unique type.
+    ///
+    /// For example the type of `a` here:
+    ///
+    /// ```rust
+    /// fn foo() -> i32 { 1 }
+    ///
+    /// fn hello() {
```

Comment:
I think you can get away with just writing it on two lines, which saves a little space:
```rust
fn foo() -> i32 { 1 }
let bar = foo; // bar: fn() -> i32 {foo}
```